### PR TITLE
Fix for indefinitely converging privileges when granted all and/or on all databases

### PR DESCRIPTION
**Pull request by takuvata**
Fixes Issue #155.

When database_mysql_user resource grants privileges on all databases and/or grants all privileges on particular one, checking if current privileges are correct doesn't work. This causes resource to converge on every chef run. Proposed fix makes mentioned cases to be handled correctly.

### CODE REVIEW

File: libraries/provider_database_mysql_user.rb [hunk 1]
```diff
@@ -94,20 +94,42 @@ def ishash?
...
+            :alter,
+            :create_tmp_table,
+            :lock_tables,
+            :create_view,
+            :show_view,
+            :create_routine,
+            :alter_routine,
+            :execute,
+            :event,
+            :trigger
```
Comment (ovaistariq) [inline libraries/provider_database_mysql_user.rb]: You are missing a few privileges here:
            :reload,
            :shutdown,
            :process,
            :file,
            :grant,
            :show_db,
            :super,
            :repl_slave,
            :repl_client,
            :create_user

**Comment by RobertNorthard** [comment]
:+1: Can this be merged?

**Comment by ovaistariq** [comment]
@RobertNorthard this PR only partially fixes the issue of cookbook not being idempotent. The following PR completely fixes the issue: https://github.com/opscode-cookbooks/database/pull/161 See issue #159 and #160

**Comment by jslusher** [comment]
+1 to this! I would like to be able to grant the repl_client privilege. Thanks!

**Comment by someara** [comment]
merged #161 instead. Thanks everyone

**Pull request closed without merging**